packages/framework/presence/src/systemWorkspace.ts

```diff
@@ -149,22 +150,23 @@ class SystemWorkspaceImpl implements PresenceStatesInternal, SystemWorkspace {
 				/* order */ value.rev,
 			);
 
-			// Check new attendee against audience to see if they're currently connected
-			const isAttendeeConnected = audienceMembers.has(clientConnectionId);
+			// Check new attendee to see if they're currently connected
+			const isAttendeeConnected =
+				audienceMembers.has(clientConnectionId) ||
+				senderConnectionId === clientConnectionId ||
+				connectedAttendees.has(attendee);
 
```

Comment:
This feels like a pretty complicated state system. I feel like I'll be able to break it. :)
This line in particular needs more explanation. The other two make sense.

Reply:
Added some comments here for each condition, but willing to change if one does not seem reasonable

Reply:
The commentary that is interesting to capture is why the first two checks aren't sufficient.
And perhaps make the statement less strong. Really, we consider the attendee connected if the audience says so. And we override audience when sender id matches (which we want comments for somewhere).
I believe the last case is saying: attendee is connected if any of the connection ids for the attendee are current/connected. As we process in a list this check provides if any of the past connection ids were connected. If none are to this point then we'll set disconnected and should any later connection be found connected status will be set to connected.

Reply:
Ok yes this makes sense, will add a more detailed comment along these lines

Reply:
I wanted to also add that I did not make any assumptions based on the ordering of connection IDs received from the remote datastore. That is, I did not write the code with the guarantee that previous connection IDs will appear first, and more recent ones will appear later. The last check is not necessary if that is a guarantee that we enforce, and I think this function can be written differently.